src/main/java/kitchenpos/menu/domain/MenuProducts.java

```diff
@@ -17,27 +17,16 @@
         return Collections.unmodifiableList(menuProducts);
     }
 
-    public void validationByPrice() {
-        menuProducts.stream().forEach(menuProduct -> validationByPrice(menuProduct.getProduct(), menuProduct.getMenu()));
-    }
-
-    private void validationByPrice(Product product, Menu menu) {
-        BigDecimal totalPrice = calcTotalPrice(product);
+    public void validationByPrice(Price menuPrice) {
+        Price totalPrice = this.menuProducts.stream()
+            .map(MenuProduct::calculate)
+            .reduce(Price::plus).orElse(new Price());
```

Comment:
팩토리 메소드를 적용해서 아래 처럼 된다면 좀 더 명확할 것 같습니다 😄 
`Price()`를 확인해보지 않아도 될 것 같거든요 😄 🙇 
```suggestion
            .reduce(Price::plus).orElse(Price.zero());
```

Reply:
피드백 감사합니다!